pkg/snmp/traps/formatter_test.go

```diff
@@ -225,24 +295,179 @@ func TestNewJSONFormatterWithNilStillWorks(t *testing.T) {
+						"value": float64(7),
+					},
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.7",
+						"type":  "integer",
+						"value": float64(2),
+					},
+				},
+			},
+			expectedTags: []string{
+				"snmp_version:2",
+				"device_namespace:default",
+				"snmp_device:127.0.0.1",
+			},
+		},
+		{
+			description: "test enum variable resolution with bad variable",
+			trap:        BadValueExampleV2Trap,
+			resolver:    resolverWithData,
+			expectedContent: map[string]interface{}{
+				"snmpTrapName":  "linkUp",
+				"snmpTrapMIB":   "IF-MIB",
+				"snmpTrapOID":   "1.3.6.1.6.3.1.1.5.4",
+				"ifIndex":       float64(9001),
+				"ifAdminStatus": "test",
+				"ifOperStatus":  "lowerLayerDown",
+				"uptime":        float64(1000),
+				"variables": []interface{}{
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.1",
+						"type":  "integer",
+						"value": float64(9001),
+					},
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.8",
+						"type":  "integer",
+						"value": float64(7),
+					},
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.7",
+						"type":  "integer",
+						"value": "test",
+					},
+				},
+			},
+			expectedTags: []string{
+				"snmp_version:2",
+				"device_namespace:default",
+				"snmp_device:127.0.0.1",
+			},
+		},
+		{
+			description: "test enum variable resolution when mapping absent",
+			trap:        NoEnumMappingExampleV2Trap,
+			resolver:    resolverWithData,
+			expectedContent: map[string]interface{}{
+				"snmpTrapName":  "linkUp",
+				"snmpTrapMIB":   "IF-MIB",
+				"snmpTrapOID":   "1.3.6.1.6.3.1.1.5.4",
+				"ifIndex":       float64(9001),
+				"ifAdminStatus": float64(8),
+				"ifOperStatus":  "lowerLayerDown",
+				"uptime":        float64(1000),
+				"variables": []interface{}{
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.1",
+						"type":  "integer",
+						"value": float64(9001),
+					},
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.8",
+						"type":  "integer",
+						"value": float64(7),
+					},
+					map[string]interface{}{
+						"oid":   "1.3.6.1.2.1.2.2.1.7",
+						"type":  "integer",
+						"value": float64(8),
+					},
+				},
+			},
+			expectedTags: []string{
+				"snmp_version:2",
+				"device_namespace:default",
+				"snmp_device:127.0.0.1",
+			},
+		},
+	}
 
-	tags := formatter.GetTags(packet)
-	assert.Equal(t, tags, []string{
-		"snmp_version:2",
-		"device_namespace:default",
-		"snmp_device:127.0.0.1",
-	})
+	for _, d := range data {
+		t.Run(d.description, func(t *testing.T) {
+			formatter, err := NewJSONFormatter(d.resolver)
+			require.NoError(t, err)
+			packet := createTestPacket(d.trap)
+			data, err := formatter.FormatPacket(packet)
+			require.NoError(t, err)
+			content := make(map[string]interface{})
+			json.Unmarshal(data, &content)
+
+			// map comparisons shouldn't be reliant on ordering with this lib
+			// however variables are a slice, they must be sorted
```

Comment:
Variables should end up in the same order than they were received. I think you should be able to remove your `byOID` ordering. This way it will also make sure that the formatter keeps variables in the same order.

Reply:
Good call, removed `byOID` and the sorting from here.